repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -112,55 +81,50 @@ class HerpFilter @Inject()(configurationService: ConfigurationService) extends F
       }
     }
 
-    def doFilterOut(valuesMap: Map[String, Value]): Boolean = {
-      val hitFilters = filtersOut.filter { andFilter =>
-        andFilter.forall { case (keyOrig, pattern) =>
-          val keySplit = keyOrig.split('.')
-          val foundMatch: Boolean = valuesMap(keySplit(0)) match {
-            case LongValue(Some(x)) => pattern.findFirstIn(x.toString).isDefined
-            case StringValue(Some(x)) => pattern.findFirstIn(x).isDefined
-            case StringArrayValue(Some(x)) => x.filter { s => pattern.findFirstIn(s).isDefined}.nonEmpty
-            case MapValue(Some(x)) =>
-              // IF there is a sub key,
-              // THEN try the map;
-              // ELSE just bail.
-              if (keySplit.size > 1) {
-                x.getOrElse(keySplit(1), Array("")).filter { s => pattern.findFirstIn(s).isDefined}.nonEmpty
-              } else {
-                false
-              }
+    def filter(valuesMap: Map[String, Any]): Boolean = {
+      filtersOut.exists { fo =>
+        fo.getMatch.asScala.forall { m =>
+          val field = m.getField
+          val pattern = m.getRegex.r.pattern
```

Comment:
This should not happen on every request. Do it once when we load the configuration.